extensions/Worker.Extensions.Http.AspNetCore/src/Coordinator/ContextReference.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore
+{
+    internal class ContextReference : IDisposable
+    {
+        private readonly TaskCompletionSource<bool> _functionStartTask = new();
+        private readonly TaskCompletionSource<bool> _functionCompletionTask = new();
+
+        private TaskCompletionSource<HttpContext> _httpContextValueSource = new();
+        private TaskCompletionSource<FunctionContext> _functionContextValueSource = new();
```

Comment:
@brettsam @fabiocav Is it safe to not use TaskContinuation.RunAsynchronously here given that by default some parts of continuation will now by default be executed on the thread that called the completion source set methods? I'm currently looking into an obscure bug of a client and started wondering whether this was deliberate or an oversight.